### PR TITLE
Update powershell to support RulesEngine in Cdn

### PR DESCRIPTION
**Pull request by hytao**
## Description

Update PowerShell based on latest swagger to support RulesEngine in Cdn.

Powershell design review:
https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/301
## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/Accounts/Accounts/AzureRmAlias/Mappings.json [hunk 1]
```diff
@@ -409,7 +409,7 @@
     "Get-AzCdnCustomDomain": "Get-AzureRmCdnCustomDomain",
     "New-AzCdnCustomDomain": "New-AzureRmCdnCustomDomain",
     "Remove-AzCdnCustomDomain": "Remove-AzureRmCdnCustomDomain",
-    "Enable-AzCdnCustomDomain": "Enable-AzureRmCdnCustomDomain",
+    "Enable-AzCdnCustomDomainHttps": "Enable-AzureRmCdnCustomDomainHttps",
```
Comment (cormacpayne) [inline src/Accounts/Accounts/AzureRmAlias/Mappings.json]: @hytao please revert the change made in this file -- it won't be necessary to update this file

File: src/Cdn/Cdn/Az.Cdn.psd1 [hunk 2]
```diff
@@ -81,13 +81,14 @@ CmdletsToExport = 'Get-AzCdnProfile', 'Get-AzCdnProfileSsoUrl', 'New-AzCdnProfil
                'New-AzCdnEndpoint', 'Unpublish-AzCdnEndpointContent', 
                'Remove-AzCdnEndpoint', 'Set-AzCdnEndpoint', 'Start-AzCdnEndpoint', 
                'Stop-AzCdnEndpoint', 'Test-AzCdnCustomDomain', 
-               'Get-AzCdnCustomDomain', 'New-AzCdnCustomDomain', 
-               'Remove-AzCdnCustomDomain', 'Enable-AzCdnCustomDomain',
```
Comment (cormacpayne) [inline src/Cdn/Cdn/Az.Cdn.psd1]: @hytao why is `Enable-AzCdnCustomDomain` being removed? Removing this cmdlet would be a breaking change, which is not allowed for any of the upcoming releases.

File: src/Cdn/Cdn/CustomDomain/EnableAzureRmCdnCustomDomain.cs [hunk 3]
```diff
@@ -1,81 +0,0 @@
-﻿// ----------------------------------------------------------------------------------
```
Comment (cormacpayne) [inline src/Cdn/Cdn/CustomDomain/EnableAzureRmCdnCustomDomain.cs]: @hytao this file should not be removed

File: src/Cdn/Cdn/Models/Endpoint/PSDeliveryRuleCacheExpirationAction.cs [hunk 4]
```diff
@@ -20,6 +20,8 @@ namespace Microsoft.Azure.Commands.Cdn.Models.Endpoint
 {
     public class PSDeliveryRuleCacheExpirationAction: PSDeliveryRuleAction
     {
-        public PSCacheExpirationActionParameters Parameters { get; set; }
```
Comment (cormacpayne) [inline src/Cdn/Cdn/Models/Endpoint/PSDeliveryRuleCacheExpirationAction.cs]: @hytao is this change needed? If this `PSDeliveryRuleCacheExpirationAction` type is exposed to the user in any way (_e.g._, as an output or parameter type), then this will be a breaking change as the `Parameters` property is no longer accessible by the user.

File: src/Cdn/Cdn/Models/Endpoint/PSDeliveryRule.cs [hunk 5]
```diff
@@ -20,11 +20,13 @@ namespace Microsoft.Azure.Commands.Cdn.Models.Endpoint
 {
     public class PSDeliveryRule
     {
+        public string Name{ get; set; }
+        
         public int Order { get; set; }
 
-        public IList<PSDeliveryRuleAction> Actions { get; set; }
```
Comment (cormacpayne) [inline src/Cdn/Cdn/Models/Endpoint/PSDeliveryRule.cs]: @hytao do you need to rename the `Actions` and `Conditions` properties in this file? We should be able to keep them plural since they aren't parameters, and making this change would be a breaking change if the `PSDeliveryRule` type was exposed to the user (_e.g._, as an output or parameter type)

File: src/Cdn/Cdn/help/Az.Cdn.md [hunk 6]
```diff
@@ -20,9 +20,6 @@ Disables Custom Domain HTTPS (Deprecated).
 ### [Disable-AzCdnCustomDomainHttps](Disable-AzCdnCustomDomainHttps.md)
 Disables Custom Domain HTTPS.
 
-### [Enable-AzCdnCustomDomain](Enable-AzCdnCustomDomain.md)
```
Comment (cormacpayne) [inline src/Cdn/Cdn/help/Az.Cdn.md]: @hytao this shouldn't be removed

File: src/Cdn/Cdn/help/Enable-AzCdnCustomDomain.md [hunk 7]
```diff
@@ -1,191 +0,0 @@
----
```
Comment (cormacpayne) [inline src/Cdn/Cdn/help/Enable-AzCdnCustomDomain.md]: @hytao this file shouldn't be removed

File: src/Cdn/Cdn/help/New-AzCdnDeliveryPolicy.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
+---
+external help file: Microsoft.Azure.PowerShell.Cmdlets.Cdn.dll-Help.xml
+Module Name: Az.Cdn
+online version:
```
Comment (cormacpayne) [inline src/Cdn/Cdn/help/New-AzCdnDeliveryPolicy.md]: @hytao for each of the new markdown files you're adding, there should be a URL added for the `online version` property in the header of the file. It would be in the form of `https://docs.microsoft.com/en-us/powershell/module/az.cdn/{{cmdlet-name}}`

See [this file](https://raw.githubusercontent.com/Azure/azure-powershell/master/src/Cdn/Cdn/help/Confirm-AzCdnEndpointProbeURL.md) for an example of the URL at the top of the file.

File: src/Cdn/Cdn/help/New-AzCdnDeliveryPolicy.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+```
+
+## DESCRIPTION
+The **New-AzCdnDeliveryPolicy** cmdlet creates a delivery policy for CDN endpoint creation.
+
+## EXAMPLES
+
+### Example 1
+```powershell
+PS C:\> New-AzCdnDeliveryPolicy -Description "Sample Policy" -Rule $rule
```
Comment (cormacpayne) [inline src/Cdn/Cdn/help/New-AzCdnDeliveryPolicy.md]: @hytao for the examples you're adding in the new cmdlets, please make sure that the corresponding output that the user would see when running the cmdlet is displayed in the code block as well

File: src/Cdn/Cdn/Az.Cdn.psd1 (hunk 2 above)
Comment (hytao) [reply src/Cdn/Cdn/Az.Cdn.psd1]: This cmdlet is marked as deprecated so I think it would be better to remove it.

I could add it back if removing is not allowed

File: src/Cdn/Cdn/Models/Endpoint/PSDeliveryRuleCacheExpirationAction.cs (hunk 4 above)
Comment (hytao) [reply src/Cdn/Cdn/Models/Endpoint/PSDeliveryRuleCacheExpirationAction.cs]: It is not necessary. I can revert the change here

**Comment by hytao** [comment]
Hi, @cormacpayne . I have made an update to fix the issue. Could you review it? Thanks!

File: src/Cdn/Cdn/ChangeLog.md [hunk 10]
```diff
@@ -20,6 +20,9 @@
 
 ## Upcoming Release
 
+## Version 1.3.0
```
Comment (cormacpayne) [inline src/Cdn/Cdn/ChangeLog.md]: @hytao you can remove this `## Version 1.3.0` header and put the below line under `## Upcoming Release`

File: src/Cdn/Cdn/help/Az.Cdn.md [hunk 11]
```diff
@@ -62,6 +62,18 @@ Gets the resource usage for a susbscription.
 ### [New-AzCdnCustomDomain](New-AzCdnCustomDomain.md)
 Creates a custom domain for a CDN endpoint.
 
+### [New-AzCdnDeliveryPolicy](New-AzCdnDeliveryPolicy.md)
+{{ Fill in the Synopsis }}
```
Comment (cormacpayne) [inline src/Cdn/Cdn/help/Az.Cdn.md]: @hytao please fill in these `{{ Fill in the Synopsis }}` templates with a brief overview of the cmdlets' purpose

File: src/Cdn/Cdn/Models/Endpoint/PSDeliveryRuleUrlPathCondition.cs [hunk 12]
```diff
@@ -1,25 +0,0 @@
-﻿// ----------------------------------------------------------------------------------
```
Comment (cormacpayne) [inline src/Cdn/Cdn/Models/Endpoint/PSDeliveryRuleUrlPathCondition.cs]: @hytao was this type being used anywhere?

File: src/Cdn/Cdn/Models/Endpoint/PSUrlFileExtensionConditionParameters.cs [hunk 13]
```diff
@@ -1,26 +0,0 @@
-﻿// ----------------------------------------------------------------------------------
```
Comment (cormacpayne) [inline src/Cdn/Cdn/Models/Endpoint/PSUrlFileExtensionConditionParameters.cs]: @hytao was this type being used anywhere?

File: src/Cdn/Cdn/Models/Endpoint/PSDeliveryRuleUrlPathCondition.cs (hunk 12 above)
Comment (hytao) [reply src/Cdn/Cdn/Models/Endpoint/PSDeliveryRuleUrlPathCondition.cs]: This type was introduced but never used by customer.

File: src/Cdn/Cdn/Models/Endpoint/PSUrlFileExtensionConditionParameters.cs (hunk 13 above)
Comment (hytao) [reply src/Cdn/Cdn/Models/Endpoint/PSUrlFileExtensionConditionParameters.cs]: It was used in ModleExtensions.cs. This type has been removed from latest swagger and .net SDK

File: src/Cdn/Cdn/Models/Endpoint/PSDeliveryRuleUrlPathCondition.cs (hunk 12 above)
Comment (hytao) [reply src/Cdn/Cdn/Models/Endpoint/PSDeliveryRuleUrlPathCondition.cs]: This was not an outputType or parameter for any cmdlet. I think it should be fine to remove it.

File: src/Cdn/Cdn/Models/Endpoint/PSUrlFileExtensionConditionParameters.cs (hunk 13 above)
Comment (hytao) [reply src/Cdn/Cdn/Models/Endpoint/PSUrlFileExtensionConditionParameters.cs]: This was not an outputType or parameter for any cmdlet. I think it should be fine to remove it.
